packages/handlers/mysql/src/generate-schema.ts

```diff
@@ -0,0 +1,284 @@
+  GraphQLFieldConfigMap,
+  GraphQLInputFieldConfigMap,
+  GraphQLID,
+  GraphQLNonNull,
+  GraphQLType,
+  GraphQLSchema,
+  GraphQLEnumType,
+  GraphQLEnumValueConfigMap,
+  GraphQLList,
+} from 'graphql';
+import { BigIntResolver as GraphQLBigInt, DateTimeResolver as GraphQLDateTime } from 'graphql-scalars';
+import { camelCase, pascalCase } from 'change-case';
+import graphqlFields from 'graphql-fields';
+
+interface ColumnInfo {
+  Type: string;
+  Field: string;
+  Comment?: string;
+  Key?: 'PRI' | 'MUL';
+  Null?: 'NO' | 'YES';
+  Default?: any;
+}
+
+export class MySQLGraphQLSchemaFactory {
+  private scalarsMap = new Map([
+    ['varchar', GraphQLString],
+    ['char', GraphQLString],
+    ['tinytext', GraphQLString],
+    ['text', GraphQLString],
+    ['mediumtext', GraphQLString],
+    ['longtext', GraphQLString],
+    ['int', GraphQLInt],
+    ['tinyint', GraphQLInt],
+    ['smallint', GraphQLInt],
+    ['mediumint', GraphQLInt],
+    ['bigint', GraphQLBigInt],
+    ['float', GraphQLFloat],
+    ['double', GraphQLFloat],
+    ['real', GraphQLFloat],
+    ['decimal', GraphQLFloat],
+    ['date', GraphQLDateTime],
+    ['time', GraphQLString],
+    ['datetime', GraphQLDateTime],
+    ['timestamp', GraphQLDateTime],
+    ['year', GraphQLString],
+  ]);
+
+  private outputTypeMap = new Map<string, GraphQLObjectType>();
+
+  private queryFields: GraphQLFieldConfigMap<any, any> = {};
+  private mutationFields: GraphQLFieldConfigMap<any, any> = {};
+
+  private connection: Connection;
+  constructor(connectionString: string) {
+    this.connection = createConnection(connectionString);
+  }
+
+  private async query<T>(queryStatement: string): Promise<T[] & { insertId: string }> {
+    return new Promise((resolve, reject) => {
+      this.connection.query(queryStatement, function (error, results) {
+        if (error) {
+          return reject(error);
+        }
+        return resolve(results);
+      });
+    });
+  }
+
+  private getColumnsForTableName(tableName: string): Promise<ColumnInfo[]> {
+    return this.query(`SHOW FULL FIELDS FROM ${tableName}`);
+  }
+
+  private async getTableNames() {
+    const results = await this.query<{ TABLE_NAME: string }>(
+      `SELECT TABLE_NAME FROM information_schema.tables WHERE TABLE_SCHEMA = "employees"`
+    );
+    return results.map(result => result.TABLE_NAME);
+  }
+
+  private processColumns(tableName: string, columns: ColumnInfo[]) {
+    const createInputFields: GraphQLInputFieldConfigMap = {};
+    const updateInputFields: GraphQLInputFieldConfigMap = {};
+    const outputFields: GraphQLFieldConfigMap<any, any> = {};
+    let primaryKeyFieldName = 'id';
+    for (const column of columns) {
+      const fieldName = column.Field;
+      let columnGraphQLType: GraphQLType;
+      if (column.Key) {
+        columnGraphQLType = GraphQLID;
+        if (column.Key === 'PRI') {
+          primaryKeyFieldName = fieldName;
+        }
+      } else if (column.Type.startsWith('enum(')) {
```

Comment:
Is there a better way to do that? Instead of parsing strings? 
Because SQL has a generic AST that works on MySQL as well, and I think maybe it could be better to "visit" it instead of parsing strings.

Reply:
That is the only place we parse strings. Actually this is not generic SQL and it is MySQL specific.